builder/context.go

```diff
@@ -234,6 +237,15 @@ func getScanArgs(
 		args = append(args, "--build-arg", buildArg)
 	}
 
+	for _, cred := range credentials {
+		if serializedCredential, err := cred.String(); err != nil {
+			// Log error here, but throw it from scanner
+			log.Panicf("Credential validation failed for Source registry credential, err: %+v", err)
```

Comment:
>Panicf [](start = 7, length = 6)

Do we use panic in other places?

Reply:
So I took this from how the runner in [worker.go](https://msazure.visualstudio.com/AzureContainerRegistry/_git/Build?path=%2Fsrc%2Facr-build-runner%2Fscheduler%2Fworker.go&version=GBakashsinghal%2FbuilderArtifactSupport&line=534&lineEnd=535&lineStartColumn=1&lineEndColumn=1&lineStyle=plain) handles credential serialization. Here they log the error rather than throw it since the function it is called in doesn't return an  error. If you think its better to throw an error, I can change the `getScanArgs` method to return an error if it there is one?

Reply:
Yea I had taken this from another piece of code. I will make some changes to the code so it return an error instead.